paimon-flink/paimon-flink-1.18/src/main/java/org/apache/paimon/flink/procedure/MigrateDatabaseProcedure.java

```diff
@@ -78,11 +95,24 @@ public String[] call(
                         p,
                         ParameterUtils.parseCommaSeparatedKeyValues(properties));
 
+        int errorCount = 0;
+        int successCount = 0;
+
         for (Migrator migrator : migrators) {
-            migrator.executeMigrate();
-            migrator.renameTable(false);
+            try {
+                migrator.executeMigrate();
+                migrator.renameTable(false);
+                successCount++;
+            } catch (Exception e) {
+                errorCount++;
+                LOG.error("Call migrate_database error:" + e.getMessage());
+            }
         }
+        String retStr =
+                String.format(
+                        "migrate database is finished, success cnt: %s , failed cnt: %s",
+                        String.valueOf(successCount), String.valueOf(errorCount));
 
-        return new String[] {"Success"};
+        return new String[] {retStr};
```

Comment:
The code is the same as above, thecode should be extracted into methods.

Reply:
Done